### PR TITLE
doc: update TAIL docs to match current best practices

### LINKED ISSUE #5483: docs: improve TAIL docs and tests for new transaction behavior
We should make sure that our docs and tests for psycopg and Npgsql correctly account for the new transaction behavior. I think we're missing an `autocommit` flag on the psycopg doc and a test for `DECLARE CURSOR` with Npgsql entirely.

/cc @mjibson

### PR DESCRIPTION
**Pull request by mjibson**
Verify npgsql changes by making the smoke tests match the doc.

Fixes #5483

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5488)

### CODE REVIEW

**Comment by mjibson** [comment]
Autocommit is not needed for psycopg2 because autocommit is false by default which causes new psycopg "cursors" (different from a SQL `CURSOR`) to issue a `BEGIN`.

**Comment by mjibson** [comment]
Updated to add a warning about psycopg3 being not stable.

File: doc/user/content/sql/tail.md [hunk 1]
```diff
@@ -270,104 +272,51 @@ with conn.cursor() as cur:
...
-If you want to see the updates that had occurred in the last 30 seconds, you could run:
+dsn = "postgresql://localhost:6875/materialize?sslmode=disable"
+conn = psycopg3.connect(dsn)
 
-```sql
-TAIL some_materialized_view AS OF now() - '30s'::INTERVAL
+conn = psycopg3.connect(dsn)
+with conn.cursor() as cur:
+    for row in cur.stream("TAIL t"):
+        print(row)
```
Comment (cirego) [inline doc/user/content/sql/tail.md:299]: Would you be willing to add an `async` example as well?

**Review by cirego — APPROVED** [review]
👍🏻

File: doc/user/content/sql/tail.md (hunk 1 above)
Comment (mjibson) [reply doc/user/content/sql/tail.md:299]: Yes. I think on friday I'm going to work on language tabs. I'll add that then.

File: test/lang/csharp/SmokeTest.cs [hunk 2]
```diff
@@ -43,36 +43,36 @@ public void BasicQuery()
...
+                Assert.AreEqual(1, reader[1]); // diff
+                Assert.AreEqual(2, reader[2]); // a
+                Assert.AreEqual("b", reader[3]); // b
+                Assert.IsFalse(reader.Read());
+            }
+            txn.Commit();
 
-            // The tail won't end until we send a cancel request.
-            reader.Cancel();
+            new NpgsqlCommand("DROP TABLE t", conn).ExecuteNonQuery();
```
Comment (benesch) [inline test/lang/csharp/SmokeTest.cs:75]: I’d like to keep the binary COPY test. If you happen to get to it, great! If not I’ll grab it when I get home.

**Pull request merged**